tests/php/src/Cli/Export/ExportOptions.php

```diff
@@ -0,0 +1,359 @@
+		global $wpdb;
+
+		if ( empty( $keys ) ) {
+			return [];
+		}
+
+		$query = sprintf(
+			'SELECT `option_name`, `option_value` FROM `%s` WHERE `option_name` IN (%s);',
+			$wpdb->options,
+			implode(
+				',',
+				array_map( static function ( $key ) { return "'" . esc_sql( $key ) . "'"; }, $keys )
+			)
+		);
+
+		$options = array_filter( (array) $wpdb->get_results( $query, 'ARRAY_A' ) );
+
+		return array_combine(
+			array_column( $options, 'option_name' ),
+			array_map( 'maybe_unserialize', array_column( $options, 'option_value' ) )
+		);
+	}
+
+	/**
+	 * Adapt the options to get rid of hard-coded elements like IDs.
+	 *
+	 * @param array        $options       Associative array of options to adapt.
+	 * @param ExportResult $export_result Export result to adapt.
+	 * @return array Adapted associative array of options.
+	 */
+	private function adapt_options( $options, $export_result ) {
+		foreach ( $options as $key => $value ) {
+			switch ( $key ) {
+				case 'woocommerce_shop_page_title':
+				case 'woocommerce_cart_page_title':
+				case 'woocommerce_checkout_page_title':
+				case 'woocommerce_myaccount_page_title':
+				case 'woocommerce_edit_address_page_title':
+				case 'woocommerce_view_order_page_title':
+				case 'woocommerce_change_password_page_title':
+				case 'woocommerce_logout_page_title':
+					// @TODO
+					break;
+
+				case 'page_for_posts':
+				case 'page_on_front':
+					$options[ $key ] = $this->get_page_title_from_post_id( $value );
+					break;
+
+				case 'nav_menu_locations':
+					$options[ $key ] = $this->get_nav_menu_location( $value );
+					break;
+
+				case 'woocommerce_product_cat':
+					// @TODO
+					break;
+
+				case 'custom_logo':
+					$media_uploader = new MediaFileUploader();
+					$options[ $key ] = $media_uploader->upload(
+						$export_result->get_site_name(),
+						$this->get_attachment_url_from_attachment_id( $value )
+					);
+					break;
+			}
+		}
+
+		return $options;
+	}
+
+	/**
+	 * Skip the options that are marked as excluded.
+	 *
+	 * @param string $option_key Option key to check.
+	 * @return bool Whether to keep the option.
+	 */
+	private function skip_excluded_options( $option_key ) {
+		foreach ( self::EXCLUDED_OPTIONS as $option_pattern ) {
+			if ( 1 === preg_match( $option_pattern, $option_key ) ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Skip the options that still have their default value.
+	 *
+	 * @param string $option_value Option value to check.
+	 * @param string $option_key   Option key to check.
+	 * @return bool Whether to keep the option.
+	 */
+	private function skip_default_values( $option_value, $option_key ) {
+		if ( ! array_key_exists( $option_key, self::OPTION_DEFAULTS ) ) {
+			return true;
+		}
+
+		return self::OPTION_DEFAULTS[ $option_key ] !== $option_value;
+	}
+
+	/**
+	 * Get the post title from a post ID.
+	 *
+	 * @param string|int $post_id ID of the post to get the title from.
```

Comment:
```suggestion
	 * @param int $post_id ID of the post to get the title from.
```

Reply:
We're talking about WordPress here... But I can cast to an integer when I call this method to enforce an int.

Reply:
Resolved via https://github.com/ampproject/amp-wp/pull/5162/commits/05d9be2779751fc870d8addacae5b3d09347d64b